webapp/javascript/components/TimelineChart/TimelineChartWrapper.tsx

```diff
@@ -212,27 +213,36 @@ class TimelineChartWrapper extends React.Component<
     ) {
       const newFlotOptions = this.state.flotOptions;
       newFlotOptions.grid.markings = this.plotMarkings();
+      newFlotOptions.annotations = this.composeAnnotationsList();
       this.setState({ flotOptions: newFlotOptions });
     }
   }
 
+  composeAnnotationsList = () => {
+    return Array.isArray(this.props.annotations)
+      ? this.props.annotations?.map((a) => ({
+          timestamp: a.timestamp,
+          content: a.content,
+          type: 'message',
+          color: ANNOTATION_COLOR,
+        }))
+      : null;
+  };
```

Comment:
Already said somewhere else, but please don't add a third state to an array. An array is either empty or non-empty. If you need to have another state make it clear with another variable.